File: Backend/MetadataService/MetadataService.Infrastructure/Models/Audio.cs
﻿using System;
using System.Collections.Generic;

namespace MetadataService.Infrastructure.Models;

public partial class Audio
{
    public int Id { get; set; }

    public string? Title { get; set; }

    public string? Description { get; set; }

    public byte[]? Thumbnail { get; set; }

    public int DurationSec { get; set; }

    public long? FileSize { get; set; }

    public string? FileType { get; set; }

    public byte[]? FileChecksum { get; set; }

    public int? Listens { get; set; }

    public int? StatusId { get; set; }

    public int? VisibilityId { get; set; }

    public DateTime? UploadedAt { get; set; }

    public int? UploaderId { get; set; }

    public virtual ICollection<Listen> ListensNavigation { get; set; } = new List<Listen>();

    public virtual Status? Status { get; set; }

    public virtual ICollection<Tag> Tags { get; set; } = new List<Tag>();

    public virtual Visibility? Visibility { get; set; }
}

File: Frontend/audiowave6/lib/domain/entities/audio.dart
import 'dart:typed_data';

class Audio {
  final int id;
  final String? title;
  final String? description;
  final Uint8List? thumbnail;
  final int durationSec;
  final int? fileSize;
  final String? fileType;
  final Uint8List? fileChecksum;
  final int? listens;
  final int? statusId;
  final int visibilityId;
  final DateTime? uploadedAt;
  final int? uploaderId;
  final List<String>? tags;

  Audio({
    required this.id,
    this.title,
    this.description,
    this.thumbnail,
    required this.durationSec,
    this.fileSize,
    this.fileType,
    this.fileChecksum,
    this.listens,
    this.statusId,
    required this.visibilityId,
    this.uploadedAt,
    this.uploaderId,
    this.tags,
  });
}
